text_reading/src/main/scala/org/clulab/aske/automates/OdinActions.scala

```diff
@@ -124,28 +124,31 @@ class OdinActions(val taxonomy: Taxonomy, expansionHandler: Option[ExpansionHand
 
   def looksLikeAVariable(mentions: Seq[Mention], state: State): Seq[Mention] = {
     //returns mentions that look like a variable
-    for {
-      m <- mentions
-      varMention = m match {
-        case tb: TextBoundMention => m
-        case rm: RelationMention => m.arguments.getOrElse("variable", Seq()).head
-        case em: EventMention => m.arguments.getOrElse("variable", Seq()).head
-        case _ => ???
-      }
-      if varMention.words.length == 1
-      word = varMention.words.head
-      if word.length <= 6
-      if (word.toLowerCase != word // mixed case or all UPPER
+    def passesFilters(v: Mention, isArg: Boolean): Boolean = {
+      if ((v matches "Variable") && isArg) return true
```

Comment:
@BeckySharp, the condition in line 128 does not work (a test broke) because I also need this check when the Variable label is actually assigned, e.g., the word 'for' is found as a Variable by the rule 'var_calculation' because the word 'calculation' is frequently preceded by a variable (e.g., _Kcbmin calculation is implemented by..._), then the action looksLikeAVariable is run, and I know that it's not really a Variable. Does that make sense? I got rid of the condition.

Reply:
ah, poop.
Ok, I'll modify the taxonomy and add an entry for Variables found with a Gazetteer, which i need to pass everytime...

Reply:
Oh, I see what you were going for with that...